src/transports/http.c

```diff
@@ -359,6 +360,9 @@ static int on_headers_complete(http_parser *parser)
 					t->parse_error = PARSE_ERROR_REPLAY;
 					return 0;
 				}
+			} else {
+				giterr_set(GITERR_NET, "authentication failed, incorrect credentials");
```

Comment:
Typically a 403 is returned for authorization errors, but I can't speak to what every Git server out there does, so I don't mind just "authentication failed".